Sources/BrowserServicesKit/Navigation/NavigationAction.swift

```diff
@@ -0,0 +1,220 @@
+            && lhs.request.isEqual(to: rhs.request)
+    }
+
+}
+
+public extension NavigationAction {
+
+    var isForMainFrame: Bool {
+        targetFrame.isMainFrame
+    }
+
+    var isTargetingNewWindow: Bool {
+        sourceFrame.identity.webView != targetFrame.identity.webView
+    }
+
+    var url: URL {
+        request.url ?? NSURL() as URL
+    }
+
+}
+
+private extension URLRequest {
+    func isEqual(to other: URLRequest) -> Bool {
+        url == other.url && httpMethod == other.httpMethod && (allHTTPHeaderFields ?? [:]) == (other.allHTTPHeaderFields ?? [:])
+            && cachePolicy == other.cachePolicy && timeoutInterval == other.timeoutInterval
+    }
+}
+
+public struct NavigationPreferences: Equatable {
+
+    public var userAgent: String?
+    public var contentMode: WKWebpagePreferences.ContentMode
+
+    fileprivate var javaScriptEnabledValue: Bool
+    @available(macOS 11.0, iOS 14.0, *)
+    public var javaScriptEnabled: Bool {
+        get {
+            javaScriptEnabledValue
+        }
+        set {
+            javaScriptEnabledValue = newValue
+        }
+    }
+
+    public static let `default` = NavigationPreferences(userAgent: nil, contentMode: .recommended, javaScriptEnabled: true)
+
+    public init(userAgent: String?, contentMode: WKWebpagePreferences.ContentMode, javaScriptEnabled: Bool) {
+        self.userAgent = userAgent
+        self.contentMode = contentMode
+        self.javaScriptEnabledValue = javaScriptEnabled
+    }
+
+    internal init(userAgent: String?, preferences: WKWebpagePreferences) {
+        self.contentMode = preferences.preferredContentMode
+        if #available(macOS 11.0, iOS 14.0, *) {
+            self.javaScriptEnabledValue = preferences.allowsContentJavaScript
+        } else {
+            self.javaScriptEnabledValue = true
+        }
+    }
+
+    internal func applying(to preferences: WKWebpagePreferences) -> WKWebpagePreferences {
+        preferences.preferredContentMode = contentMode
+        if #available(macOS 11.0, iOS 14.0, *) {
+            preferences.allowsContentJavaScript = javaScriptEnabled
+        }
+        return preferences
+    }
+
+}
+
+public enum NavigationActionPolicy {
+    case allow
+    case cancel(with: NavigationActionCancellationRelatedAction)
+    case download
+
+    public static var cancel: NavigationActionPolicy = .cancel(with: .none)
+}
+
+extension NavigationActionPolicy? {
+    /// Pass decision making to next responder
+    public static let next = NavigationActionPolicy?.none
+}
+
+public enum NavigationActionCancellationRelatedAction: Equatable {
+    case none
+    case taskCancelled
+    case redirect(URLRequest)
+    case other(UserInfo)
+}
+
+extension WKNavigationActionPolicy {
+    static let download = WKNavigationActionPolicy(rawValue: Self.allow.rawValue + 1) ?? .cancel
```

Comment:
If we bump deployment target to or above macOS 11.3, this will stop compiling. I’ve had problems with it on the App Store branch, where App Store app targets macOS 12.3.

I solved it by removing this definition completely, and using `WKNavigationActionPolicy.init(rawValue:)` initializer directly where download policy is needed.

Alternatively, you could rename it e.g. like this:

```swift
static let downloadPolicy: WKNavigationActionPolicy = {
    if #available(macOS 11.3, *) {
        return .download
    }
    return WKNavigationActionPolicy(rawValue: Self.allow.rawValue + 1) ?? .cancel
}()
```